src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java

```diff
@@ -855,12 +855,20 @@ private void collectFileAndDirectories(BitbucketSCMFile parent, List<Map> values
     public InputStream getFileContent(BitbucketSCMFile file) throws IOException, InterruptedException {
         List<String> lines = new ArrayList<>();
         int start=0;
+        String ref;
+        if(file.getRef().matches("PR-\\d+")) {  // because JENKINS-48737
+            String prId = file.getRef().replace("PR-", "");
+            ref = "refs/pull-requests/" + prId + "/from";
```

Comment:
Could respect the pull request merge strategy set on the build configuration

Reply:
agree with @casz, could we use `pull-requests/id/merge` as ref for merge strategy?

Reply:
indeed, which I was hinting at 👍

Reply:
Thanks for the feedback. I will pick up your comments. Hopefully I have time this week. :+1: